opm/autodiff/NonlinearSolver_impl.hpp

```diff
@@ -103,11 +103,11 @@ namespace Opm
             }
             linIters += report.linear_iterations;
             ++iteration;
-        } while ( (!converged && (iteration <= maxIter())) || (minIter() > iteration));
+        } while ( (!converged && (iteration <= maxIter())) || (iteration <= minIter()));
```

Comment:
It is correct that we will call nonlinearIteration() at least twice when minIter() is 1. However, the last call to nonlinearIteration() is always trivial (in this context: not solving a linear system), because it will find after assembly that the residual is low enough and claim convergence. However, this should not count as an iteration, and therefore the logic looks like it's off by one. Same for the reporting below (where we report `iteration - 1`). In the old implementation this was handled by having the initial call to assemble() outside the iteration loop, that approach does not work for more general (split for example) models.

It gets more confusing because of the minIter though. I wish we did not need that feature!